terraform/region/kms.tf

```diff
@@ -1,7 +1,79 @@
+data "aws_kms_key" "access_log_key" {
+  key_id = "alias/mrk_access_logs_lb_encryption_key-${terraform.workspace}"
+}
+
+data "aws_iam_policy_document" "s3_loadbalancer_kms" {
+
+  statement {
+    sid       = "Enable Root account permissions on Key"
+    effect    = "Allow"
+    actions   = ["kms:*"]
+    resources = ["*"]
+
+    principals {
+      type = "AWS"
+      identifiers = [
+        "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root",
+      ]
+    }
+  }
+
+  statement {
+    sid       = "Key Administrator"
+    effect    = "Allow"
+    resources = ["*"]
+    actions = [
+      "kms:Create*",
+      "kms:Describe*",
+      "kms:Enable*",
+      "kms:List*",
+      "kms:Put*",
+      "kms:Update*",
+      "kms:Revoke*",
+      "kms:Disable*",
+      "kms:Get*",
+      "kms:Delete*",
+      "kms:TagResource",
+      "kms:UntagResource",
+      "kms:ScheduleKeyDeletion",
+      "kms:CancelKeyDeletion"
+    ]
+
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/breakglass"]
+    }
+  }
+
+  statement {
+    sid    = "Allow ELB to use Key for Encryption"
+    effect = "Allow"
+
+    actions = [
+      "kms:Encrypt",
+      "kms:Decrypt",
+      "kms:ReEncrypt*",
+      "kms:GenerateDataKey*",
```

Comment:
Does the ELB need to be able to describe keys and generate data keys? In what context would this happen?

Reply:
GenerateDataKey : https://docs.aws.amazon.com/kms/latest/APIReference/API_GenerateDataKey.html - is required for the way that keys are generated, as there is a master key (CMK) and a data key which encrypts over the top, outside of KMS. this is usually stored with the data.
DescribeKey - https://docs.aws.amazon.com/kms/latest/APIReference/API_DescribeKey.html - allows the Principal to get information about a key.